### PR TITLE
refactor(cardinal): Replace the layout struct with a slice of components

### PR DESCRIPTION
**Pull request by jerargus**
Closes: #WORLD-325

## What is the purpose of the change

Remove the Layout struct. A Layout is just a slice of components, and there was no real logic attached to the struct. Using slices of components make our code more clear. In addition, we already have a concept of a collection of Components (Archetypes). 

## Testing and Verifying

Existing tests verify this is a non-functional change.

### CODE REVIEW

File: cardinal/ecs/cql/cql.go [hunk 1]
```diff
@@ -142,7 +142,7 @@ var internalCQLParser = participle.MustBuild[cqlTerm]()
 // TODO: Value is sum type is represented as a product type. There is a case where multiple properties are filled out.
 // Only one property may not be nil, The parser should prevent this from happening but for safety this should eventually
 // be checked.
-func valueToLayoutFilter(value *cqlValue, stringToComponent func(string) (component.IComponentType, bool)) (filter.LayoutFilter, error) {
+func valueToLayoutFilter(value *cqlValue, stringToComponent func(string) (component.IComponentType, bool)) (filter.ComponentFilter, error) {
```
Comment (pyrofolium) [inline cardinal/ecs/cql/cql.go:145]: Rename the function too.

File: cardinal/ecs/cql/cql.go [hunk 2]
```diff
@@ -182,19 +182,19 @@ func valueToLayoutFilter(value *cqlValue, stringToComponent func(string) (compon
 	}
 }
 
-func factorToLayoutFilter(factor *cqlFactor, stringToComponent func(string) (component.IComponentType, bool)) (filter.LayoutFilter, error) {
+func factorToLayoutFilter(factor *cqlFactor, stringToComponent func(string) (component.IComponentType, bool)) (filter.ComponentFilter, error) {
 	return valueToLayoutFilter(factor.Base, stringToComponent)
 }
```
Comment (pyrofolium) [inline cardinal/ecs/cql/cql.go:187]: Rename for all these functions?

File: cardinal/ecs/cql/cql.go (hunk 1 above)
Comment (jerargus) [reply cardinal/ecs/cql/cql.go]: 👍 my initial search for "layout" was case sensitive. Instances of "Layout" have been removed.

**Review by pyrofolium — APPROVED** [review]

**Review by pyrofolium — APPROVED** [review]
Random thought: maybe we should prefix all Interfaces with "I" just to make everything clear.

**Review by technicallyty — APPROVED** [review]

File: cardinal/ecs/cql/cql.go (hunk 2 above)
Comment (jerargus) [reply cardinal/ecs/cql/cql.go:187]: done.

**Pull request merged**